code/modules/projectiles/ammunition/boxes.dm

```diff
@@ -1,10 +1,47 @@
 /obj/item/ammo_magazine/speedloader
-	icon_state = "spdloader_magnum"
+	icon_state = "world"
 	caliber = CALIBER_PISTOL_MAGNUM
 	ammo_type = /obj/item/ammo_casing/pistol/magnum
 	material = MAT_STEEL
 	max_ammo = 6
-	multiple_sprites = 1
+	on_mob_icon = 'icons/obj/ammo/speedloader.dmi'
+	var/list/global/bullet_offsets = list(
+		list(x = 0, y = 0),
+		list(x = -2, y = -3),
+		list(x = -2, y = -7),
+		list(x = 0, y = -10),
+		list(x = 2, y = -7),
+		list(x = 2, y = -3)
+	)
+
+/obj/item/ammo_magazine/speedloader/on_update_icon()
+	cut_overlays()
+	if(length(stored_ammo))
+		var/list/overlays_to_add = list()
```

Comment:
There's not much point using an overlays list with `add_overlay()`.

Reply:
fixed